src/MergeContext.js

```diff
@@ -954,8 +959,18 @@ class PullRequest {
         return false;
     }
 
+    /// whether this PR has or had a staged commit with failed checks
+    _stagedStatusesFailed() {
+        return this._stagedStatuses === null ?
+            this._stagingFailedSomeTimeAgo : this._stagedStatuses.failed();
```

Comment:
This looks a bit inconsistent with the other code -- sometimes we update _stagingFailedSomeTimeAgo after _getStagingStatuses() and sometimes we do not. There are also calls to this._stagedStatuses.failed() that do not go through this function.

I think we should rename _stagingFailedSomeTimeAgo to _stagingFailed and make sure it is _always_ in sync with _stagedStatuses (when they are available). After that, we can drop this function and simply check _stagingFailed directly.

Reply:
Renamed.

> always in sync with _stagedStatuses

To keep such synchronization I moved this flag assignment into _getStagingStatuses().